course-material/intro-to-sql/combining-data-from-multiple-tables.md

```diff
@@ -36,4 +49,210 @@ output = "expanded_data"
+
+```{code-cell} ipython3
+%%sql 
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.account AS a
+INNER JOIN s1.district AS d 
+ON a.district_id = d.district_id
+INNER JOIN s1.link AS l 
+ON l.account_id = a.account_id
+INNER JOIN s1.card AS c 
+ON c.disp_id = l.disp_id; 
+```
+
+The result of the query above is a table containing the primary key columns from each table. The number of rows is 892, which signifies that there are 892 matching records between the four tables. This means that there are 892 customers who have accounts, cards, and are linked to a district. Next, attempt the question below to drive home your understanding of the `INNER JOIN`!
+
+#### Question 1 (Medium):
+Suppose we want to analyze the spending patterns of customers across different districts based on their account and card information. Unmatched records could represent customers who have recently opened accounts or received new cards and are yet to make any transactions. Write a SQL query to join the `account`, `card`, `district`, and `link` tables from the `s1` schema in the DuckDB database. Include only primary key columns from each table in the result set.
+
+<!-- #region -->
+<details>
+
+<summary>Answers</summary>
+
+We should use an `INNER JOIN` to exclude unmatched data points, so that we can focus our analysis on customers who have actively used their accounts and cards for transactions.
+
+Recall that in the [previous tutorial](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#what-is-a-primary-key-and-a-foreign-key), we learned that the `account` table contains a column named `district_id`, which serves as a foreign key referencing the primary key in the `district` table. Similarly, the `card` table has a column named `disp_id`, which is a foreign key referencing the primary key in the `link` table. By utilizing these relationships, we can perform a join operation on these tables to obtain a consolidated result set.
+
+To combine the `account`, `card`, `district`, and `link` tables, we can use the following SQL query:
+
+```{code-cell} ipython3
+%%sql 
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.account AS a
+INNER JOIN s1.district AS d 
+ON a.district_id = d.district_id
+INNER JOIN s1.link AS l 
+ON l.account_id = a.account_id
+INNER JOIN s1.card AS c 
+ON c.disp_id = l.disp_id; 
+```
+
+In this query, we are performing a series of `INNER JOIN` operations to merge the tables based on the specified join conditions. The `ON` clause defines the relationship between the columns that are used for joining. The `SELECT` statement retrieves only the primary keys from each of the joined tables in the result set.
+
+```{tip}
```

Comment:
Similar as above, please replace with the important flag, change the format 

the build doesn't show the code appropriately when using ~~~